server/list.js

```diff
@@ -115,35 +115,38 @@ async function fetchAllFiles({nextPageToken: pageToken, listSoFar = [], parentId
   const {data} = await drive.files.list(options)
 
   const {files, nextPageToken} = data
-  const combined = listSoFar.concat(files)
+  const nextListSoFar = listSoFar.concat(files)
 
   // If there is more data the API has not returned for the query, the request needs to continue
   if (nextPageToken) {
     return fetchAllFiles({
+      drive,
+      driveType,
+      nextListSoFar,
       nextPageToken,
-      listSoFar: combined,
-      drive
+      nextParentIds: parentIds
```

Comment:
this was the underlying issue, when paginating through a large folder, parentIds wasn't being propogated so we would jump back up to the root and the request would fail